From: Colin Watson <cjwatson@debian.org>
Date: Sun, 18 Aug 2024 12:01:35 +0100
Subject: Use local Sphinx inventories where available

---
 docs/conf.py | 33 +++++++++++++++++++--------------
 1 file changed, 19 insertions(+), 14 deletions(-)

diff --git a/docs/conf.py b/docs/conf.py
index a2df0b1..d35ae2a 100644
--- a/docs/conf.py
+++ b/docs/conf.py
@@ -70,20 +70,25 @@ try:
 except ImportError:
     pass
 
-
-intersphinx_mapping = {
-    "pytest": ("http://docs.pytest.org/en/latest/", None),
-    "python": ("http://docs.python.org/3", None),
-    "multidict": ("https://multidict.readthedocs.io/en/stable/", None),
-    "propcache": ("https://propcache.aio-libs.org/en/stable", None),
-    "yarl": ("https://yarl.readthedocs.io/en/stable/", None),
-    "aiosignal": ("https://aiosignal.readthedocs.io/en/stable/", None),
-    "aiohttpjinja2": ("https://aiohttp-jinja2.readthedocs.io/en/stable/", None),
-    "aiohttpremotes": ("https://aiohttp-remotes.readthedocs.io/en/stable/", None),
-    "aiohttpsession": ("https://aiohttp-session.readthedocs.io/en/stable/", None),
-    "aiohttpdemos": ("https://aiohttp-demos.readthedocs.io/en/latest/", None),
-    "aiojobs": ("https://aiojobs.readthedocs.io/en/stable/", None),
-}
+def check_object_path(key, url, path):
+    if path is not None and os.path.isfile(path):
+        return {key: (url, path)}
+    return {}
+
+intersphinx_mapping = {}
+for key, url, path in (
+    ("pytest", "http://docs.pytest.org/en/latest/", "/usr/share/doc/python-pytest-doc/html/objects.inv"),
+    ("python", "http://docs.python.org/3", "/usr/share/doc/python3/html/objects.inv"),
+    ("multidict", "https://multidict.readthedocs.io/en/stable/", None),
+    ("yarl", "https://yarl.readthedocs.io/en/stable/", "/usr/share/doc/python-yarl-doc/html/objects.inv"),
+    ("aiosignal", "https://aiosignal.readthedocs.io/en/stable/", None),
+    ("aiohttpjinja2", "https://aiohttp-jinja2.readthedocs.io/en/stable/", None),
+    ("aiohttpremotes", "https://aiohttp-remotes.readthedocs.io/en/stable/", None),
+    ("aiohttpsession", "https://aiohttp-session.readthedocs.io/en/stable/", None),
+    ("aiohttpdemos", "https://aiohttp-demos.readthedocs.io/en/latest/", None),
+    ("aiojobs", "https://aiojobs.readthedocs.io/en/stable/", None),
+):
+    intersphinx_mapping.update(check_object_path(key, url, path))
 
 # Add any paths that contain templates here, relative to this directory.
 templates_path = ["_templates"]
